File: Kafka-Utils-Python/producer/common_producer.py
# -*- coding: utf-8 -*-
"""
@Author YH YR
@Time 2018/12/31 13:21
"""

from kafka import KafkaProducer


class CommonKafkaProducer:
    def __init__(self, broker_list):
        self.producer = KafkaProducer(bootstrap_servers=broker_list)

    def produce(self, topic, msg):
        self.producer.send(topic, bytes(msg, encoding='utf8'))

    def __del__(self):
        self.producer.close()


if __name__ == '__main__':
    broker_list = 'localhost:9092'
    topic = 'topic_demo'

    action = CommonKafkaProducer(broker_list)
    for i in range(1000):
        action.produce(topic, 'msg -> {0}'.format(i))

File: Kafka-Utils-Java/src/main/java/com/yhyr/producer/CommonProducer.java
package com.yhyr.producer;

import org.apache.kafka.clients.producer.KafkaProducer;
import org.apache.kafka.clients.producer.Producer;
import org.apache.kafka.clients.producer.ProducerRecord;

import java.util.Properties;
import java.util.stream.IntStream;

/**
 * @author yhyr
 * @since 2018/12/30 18:02
 */
public class CommonProducer {
    public static void main(String[] args) {
        String topic = "topic_demo";
        String brokerList = "localhost:9092";

        Properties props = new Properties();
        props.put("bootstrap.servers", brokerList);
        props.put("acks", "all");
        props.put("retries", 0);
        props.put("batch.size", 16384);
        props.put("linger.ms", 1);
        props.put("buffer.memory", 33554432);
        props.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer");
        props.put("value.serializer", "org.apache.kafka.common.serialization.StringSerializer");

        Producer<String, String> producer = new KafkaProducer<>(props);

        IntStream.range(0, 1000).<ProducerRecord<String, String>>mapToObj(
            i -> new ProducerRecord<>(topic, "msg -> " + i)).forEach(producer::send);

        System.out.println("send message over");
    }
}
